model/reliability/db.go

```diff
@@ -52,41 +116,18 @@ func (filter TaskReliabilityFilter) BuildMatchStageForTask(boundaries []time.Tim
 	if filter.StartAt != nil {
 		match["$or"] = filter.BuildTaskPaginationOrBranches()
 	}
-
 	return bson.M{"$match": match}
 }
 
-// dateBoundaries returns the date boundaries when splitting the period between 'start' and 'end' in groups of 'numDays' days.
-// The boundaries are the start dates of the periods of 'numDays' (or less for the last period), starting with 'start'.
-func dateBoundaries(start time.Time, end time.Time, numDays int) []time.Time {
-	if numDays <= 0 {
-		numDays = 1
-	}
-
-	start = util.GetUTCDay(start)
-	end = util.GetUTCDay(end)
-
-	boundaries := []time.Time{}
-	duration := 24 * time.Hour * time.Duration(numDays)
-	duration = -1 * duration
-	boundary := end.Add(24 * time.Hour)
-
-	for boundary.After(start) {
-		boundaries = append(boundaries, boundary)
-		boundary = boundary.Add(duration)
-	}
-	boundaries = append(boundaries, boundary)
-	return boundaries
-}
-
 // buildDateStageGroupID builds the date of the grouped
 // period the stats document belongs in.
-func buildDateStageGroupID(fieldName string, inputDateFieldName string, start time.Time, end time.Time, numDays int) interface{} {
+func (filter TaskReliabilityFilter) buildDateStageGroupID(fieldName string, inputDateFieldName string) interface{} {
```

Comment:
why does this return an interface{}

Reply:
When group_num_days is 1, then the data can be read straight from the rolled up collection so the return value is just the field name.

When group_num_days > 1 then the aggregation must group the individual days so the return is an array of bson.M values.